### PR TITLE
SALTO-1931: added references to board

### PR DESCRIPTION
**Pull request by alonstern**
Added reference from board to fields

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/board.ts [hunk 1]
```diff
@@ -45,6 +45,11 @@ const filter: FilterCreator = () => ({
       .forEach(instance => {
         instance.value.filterId = instance.value.config?.filter?.id
         delete instance.value.config?.filter
+
+        const rankFieldId = instance.value.config?.ranking?.rankCustomFieldId
```
Comment (adamro) [inline packages/jira-adapter/src/filters/board.ts:49]: We don't need to do any work here pre+postDeploy with the changes we are doing?

File: packages/jira-adapter/src/filters/board.ts [hunk 2]
```diff
@@ -45,6 +45,11 @@ const filter: FilterCreator = () => ({
...
+        if (rankFieldId !== undefined && !rankFieldId.toString().startsWith('customfield_')) {
```
Comment (adamro) [inline packages/jira-adapter/src/filters/board.ts:50]: Also, this feels awfully fragile, is there documentation for this issue? 
Are we sure there won't be other prefix's and we will have `customfield_` to something which already has a different prefix? There is no "legit" info that returns here without this specific prefix?

Comment (alonstern) [reply packages/jira-adapter/src/filters/board.ts:50]: The issue here is that fields id is of the form `customfield_<someNumber>`. Specifically here, the API returns just the number so in order to be able to create a reference from it I need to have the whole id.
The API currently returns a typeof number so I don't think there might be another prefix.
I will add some doc to it

File: packages/jira-adapter/src/filters/board.ts (hunk 1 above)
Comment (alonstern) [reply packages/jira-adapter/src/filters/board.ts:49]: Nope, it is not deployable, I will doc it

**Review by adamro — APPROVED** [review]

**Pull request merged**